### PR TITLE
DevWs for SHINA SYSTEM containing containing Zigbee Power Meter

### PR DESCRIPTION
**Pull request by shinasys**

### CODE REVIEW

File: devicetypes/shinasys/sihas-zigbee-power-meter.src/sihas-zigbee-power-meter.groovy [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+def getATTRIBUTE_VOLTAGE() { 0x0505 }
+def getATTRIBUTE_CURRENT() { 0x0508 }
+def getATTRIBUTE_POWERFACTOR() { 0x0510 }
+def getTEMPERATURE_MEASUREMENT_MEASURED_VALUE_ATTRIBUTE() { 0x0000 }
+
+def convertHexToInt24Bit(value) {
+	int result = zigbee.convertHexToInt(value)
+    if(result & 0x800000) result |= 0xFF000000
+    return result
+}
```
Comment (KKlimczukS) [inline devicetypes/shinasys/sihas-zigbee-power-meter.src/sihas-zigbee-power-meter.groovy]: Please reformat this function, fix indentations and enclose that 'if' statement with brackets.

File: devicetypes/shinasys/sihas-zigbee-power-meter.src/sihas-zigbee-power-meter.groovy [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+def refresh() {
+	log.debug "refresh "
+	zigbee.simpleMeteringPowerRefresh() +
+		   zigbee.readAttribute(zigbee.SIMPLE_METERING_CLUSTER, ATTRIBUTE_READING_INFO_SET) + 
+		   zigbee.readAttribute(zigbee.ELECTRICAL_MEASUREMENT_CLUSTER, ATTRIBUTE_FREQUENCY) +
+		   zigbee.readAttribute(zigbee.ELECTRICAL_MEASUREMENT_CLUSTER, ATTRIBUTE_VOLTAGE) +
+		   zigbee.readAttribute(zigbee.ELECTRICAL_MEASUREMENT_CLUSTER, ATTRIBUTE_CURRENT) +
+		   zigbee.readAttribute(zigbee.ELECTRICAL_MEASUREMENT_CLUSTER, ATTRIBUTE_POWERFACTOR) +
+		   zigbee.readAttribute(zigbee.TEMPERATURE_MEASUREMENT_CLUSTER, TEMPERATURE_MEASUREMENT_MEASURED_VALUE_ATTRIBUTE)
```
Comment (KKlimczukS) [inline devicetypes/shinasys/sihas-zigbee-power-meter.src/sihas-zigbee-power-meter.groovy]: Please fix indentations here (1 tab should be enough).

File: devicetypes/shinasys/sihas-zigbee-power-meter.src/sihas-zigbee-power-meter.groovy [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	sendEvent(name: "checkInterval", value: 2 * 60 + 10 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+
+	log.debug "Configuring Reporting"
+	configCmds = zigbee.simpleMeteringPowerConfig() +
+				 zigbee.configureReporting(zigbee.SIMPLE_METERING_CLUSTER, ATTRIBUTE_READING_INFO_SET, DataType.UINT48, 5, 600, 1) +
+				 zigbee.configureReporting(zigbee.ELECTRICAL_MEASUREMENT_CLUSTER, ATTRIBUTE_FREQUENCY, DataType.UINT16, 10, 600, 3) + /* 3 unit : 0.3Hz */
+				 zigbee.configureReporting(zigbee.ELECTRICAL_MEASUREMENT_CLUSTER, ATTRIBUTE_VOLTAGE, DataType.UINT16, 5, 600, 3) + /* 3 unit : 0.3V */
+				 zigbee.configureReporting(zigbee.ELECTRICAL_MEASUREMENT_CLUSTER, ATTRIBUTE_CURRENT, DataType.UINT16, 5, 600, 1) + /* 1 unit : 0.01A */
+				 zigbee.configureReporting(zigbee.ELECTRICAL_MEASUREMENT_CLUSTER, ATTRIBUTE_POWERFACTOR, DataType.INT8, 10, 600, 1) + /* 1 unit : 0.1% */
+				 zigbee.configureReporting(zigbee.TEMPERATURE_MEASUREMENT_CLUSTER, TEMPERATURE_MEASUREMENT_MEASURED_VALUE_ATTRIBUTE, DataType.INT16, 20, 300, 10 /* 1 uint : 0.1C */)
```
Comment (KKlimczukS) [inline devicetypes/shinasys/sihas-zigbee-power-meter.src/sihas-zigbee-power-meter.groovy]: Please fix indentations here (2 tabs should be enough).

**Review by KKlimczukS — CHANGES_REQUESTED** [review]

File: devicetypes/shinasys/sihas-zigbee-power-meter.src/sihas-zigbee-power-meter.groovy [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+def getATTRIBUTE_ACTIVE_POWER() { 0x050B }
+def getATTRIBUTE_FREQUENCY() { 0x0300 }
+def getATTRIBUTE_VOLTAGE() { 0x0505 }
+def getATTRIBUTE_CURRENT() { 0x0508 }
+def getATTRIBUTE_POWERFACTOR() { 0x0510 }
+def getTEMPERATURE_MEASUREMENT_MEASURED_VALUE_ATTRIBUTE() { 0x0000 }
+
+def convertHexToInt24Bit(value) {
+	int result = zigbee.convertHexToInt(value)
+	if(result & 0x800000) {
```
Comment (KKlimczukS) [inline devicetypes/shinasys/sihas-zigbee-power-meter.src/sihas-zigbee-power-meter.groovy]: nit: add space after `if`

**Review by KKlimczukS — CHANGES_REQUESTED** [review]

**Comment by shinasys** [comment]
@greens  Are you still reviewing it?

**Review by greens — APPROVED** [review]

**Comment by greens** [comment]
lgtm, @KKlimczukS whenever you're satisfied

**Review by KKlimczukS — APPROVED** [review]

**Pull request merged**